varinfo/var_info.py

```diff
@@ -189,6 +189,38 @@ def get_variables_with_coordinates(self) -> Dict[str, OutputVariableType]:
                 if variable.references.get('coordinates') is not None
                 and not self.variable_is_excluded(variable, exclusion_pattern)}
 
+    def _is_spatial_temporal_dimension(self, dimension_path: str) -> bool:
+        """ A helper method simplifying the list comprehension in
+            `is_science_variable`.
+        """
+        dimension = self.get_variable(dimension_path)
+        return (dimension is not None
+                and (dimension.is_geographic() or dimension.is_temporal()
+                     or dimension.is_projection_x_or_y()))
```

Comment:
This is simply for readability.  This is easier to read to me, but total nit.
```suggestion
        return dimension is not None and (
            dimension.is_geographic()
            or dimension.is_temporal()
            or dimension.is_projection_x_or_y()
        )
```

Reply:
added